features/0023-did-exchange/README.md

```diff
@@ -22,61 +22,78 @@ We will explain how DIDs are exchanged, with the roles, states, and messages req
 
-In cases where both parties already possess SSI capabilities, deciding who plays the role of inviter and invitee might be a casual matter of whose phone is handier.
+The _responder_, who is the sender of an explicit invitation or the publisher of a DID with an implicit invitation, must have
+an agent capable of interacting with other agents via DIDComm.
+
+In cases where both parties already possess SSI capabilities, deciding who plays the role of _requester_ and _responder_  might be a casual matter of whose phone is handier.
 
 ### States
 
-#### null
+#### Requester
 
-No exchange exists or is in progress
+The _requester_ goes through the following states per the State Machine Tables below
 
-#### invited
+* start
+* invitation-received
+* request-sent
+* response-received
+* abandoned
+* completed
 
-The invitation has been shared with the intended _invitee_(s), and they have not yet sent a _exchange_request_.
+#### Responder
 
-#### requested
+The _responder_ goes through the following states per the State Machine Tables below
 
-A _exchange_request_ has been sent by the _invitee_ to the _inviter_ based on the information in the _invitation_.
+* start
+* invitation-sent
+* request-received
+* response-sent
+* abandoned
+* completed
 
-#### responded
+#### State Machine Tables
 
-A _exchange_response_ has been sent by the _inviter_ to the _invitee_ based on the information in the _exchange_request_.
+The following are the _requester_ and _responder_ state machines.
 
-#### complete
+The `invitation-sent` and `invitation-received` are technically outside this protocol, but are useful to show in the state machine as the invitation is the trigger to start the protocol and is referenced from the protocol as the parent thread (`pthid`).  This is discussed in more detail below.
 
-The exchange has been completed.
+The `abandoned` and `completed` states are terminal states and there is no expectation that the protocol can be continued (or even referenced) after reaching those states.
 
-![State Machine Tables](chrome_2019-01-29_07-59-38.png)
+[![State Machine Tables](did-exchange-states.png)](https://docs.google.com/spreadsheets/d/1NKfEdyebJWQGinRIAPtpXA1BVm8jgG8_i1RyeoNzzKE/edit?usp=sharing)
 
 ### Errors
 
-There are no errors in this protocol during the invitation phase. For the request and response, there are two error messages possible for each phase: one for an active rejection and one for an unknown error. These errors are sent using a **problem_report** message type specific to the DID Exchange Protocol. The following list details `problem-code`s that may be sent:
+After receiving an explicit invitation, the _requester_ may send a `problem-report` to the _responder_ using the information in the invitation to either restart the invitation process (returning to the `start` state) or to abandon the protocol. The `problem-report` may be an `Out of Band` protocol message or a DID Exchange protocol message.
```

Comment:
> The `problem-report` may be an `Out of Band` protocol message or a DID Exchange protocol message.

I don't quite follow? I'm guessing this is a typo

Reply:
I receive an OOB invitation.  I'm able to parse it and have the data necessary to send a response, but there is something else wrong with the message that I don't like.  I send back a `problem-report`.  I may send that problem report as an out-of-band message because I haven't even got to the DID Exchange part, or, I'm into DID-Exchange, but there is something I don't like about the info in the OOB message, so I send a `problem-report`.  

Does that make sense my I said the message may come under either protocol?

Reply:
Oh. You mean `problem-report` *as an adopted message of either the OOB or DIDExchange protocols*. Got it.

Reply:
Good point though.  Clarified.